### PR TITLE
feat(odyssey-react): update Modal to use scss module

### PR DESCRIPTION
**Pull request by arnoldsandoval-okta**
[OKTA-416112](https://oktainc.atlassian.net/browse/OKTA-416112)

### CODE REVIEW

File: packages/odyssey-react/src/components/Modal/Modal.module.scss [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  z-index: 10;
+  inset-block-start: 0;
+  inset-inline-end: 0;
+  inset-block-end: 0;
+  inset-inline-start: 0;
+  align-items: center;
+  justify-content: center;
+  background: rgba(cv('gray', '900'), 0.75);
+
+  .root[aria-hidden='false'] & {
```
Comment (edburyenegren-okta) [inline packages/odyssey-react/src/components/Modal/Modal.module.scss:33]: Good cleanup from the previous implementation

**Review by edburyenegren-okta — APPROVED** [review]
LGTM, thanks for logging those tickets

File: packages/odyssey-react/.storybook/main.js [hunk 2]
```diff
@@ -59,7 +59,6 @@ module.exports = {
                 @import '@okta/odyssey/src/scss/components/input-field';
                 @import '@okta/odyssey/src/scss/components/input-field-layout';
                 @import '@okta/odyssey/src/scss/components/label';
-                @import '@okta/odyssey/src/scss/components/modal';
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/.storybook/main.js:62]: `_forms-context.scss` references modal class names here too. Do you believe those are also safe to be removed now?

I think the long term strategy here will be a form `variant="compact"` or similar that can be used within parent Form components cc @edburyenegren-okta

**Review by jeffbelser-okta — APPROVED** [review]
strong work 💪 

quick inline question about contextual styles too!

File: packages/odyssey-react/src/components/Modal/Modal.module.scss (hunk 1 above)
Comment (jeffbelser-okta) [reply packages/odyssey-react/src/components/Modal/Modal.module.scss:33]: We can do even better now that we have a component and JS! These types of nested and referential selectors with higher specificity can be flattened out and simplified by applying a single class to the `dialog` and `overlay` elements that trigger these animations.

File: packages/odyssey-react/.storybook/main.js (hunk 2 above)
Comment (edburyenegren-okta) [reply packages/odyssey-react/.storybook/main.js:62]: I think we can safely drop `forms-context` here. @joejezowski-okta and I will add those styles to the new Form module if necessary.

Comment (arnoldsandoval-okta) [reply packages/odyssey-react/.storybook/main.js:62]: Removed.

File: packages/odyssey-react/src/components/Modal/Modal.module.scss (hunk 1 above)
Comment (arnoldsandoval-okta) [reply packages/odyssey-react/src/components/Modal/Modal.module.scss:33]: Will get to this as part of OKTA-419315!

**Pull request merged**